### PR TITLE
Add Preview Support for Team Review Requests

### LINKED ISSUE #678: Support preview Team Review Requests API
GitHub Announcement:
https://developer.github.com/changes/2017-07-26-team-review-request-thor-preview/

GitHub API Docs:
https://developer.github.com/v3/pulls/review_requests/

Note that the announcement describes a breaking change which may involve a bump of this package's version number.

### PR DESCRIPTION
**Pull request by RaviTezu**
Adding preview support for team review requests. 
Fixes #678

### CODE REVIEW

**Comment by RaviTezu** [comment]
As @gmlewis mentioned in the issue, this involves a breaking change and we may need to bump the package version number. 
@shurcooL

**Comment by RaviTezu** [comment]
@gmlewis @shurcooL 
I see github.go has a const `libraryVersion = "8"`. Shall I change this to `libraryVersion = "9"` ?

**Comment by shurcooL** [comment]
> I see github.go has a const `libraryVersion = "8"`. Shall I change this to `libraryVersion = "9"` ? 

Yes, please do. That'll help us communicate that there are breaking API changes being done (as a result of GitHub preview API changing).

File: github/pulls_reviewers.go [hunk 1]
```diff
@@ -13,18 +13,22 @@ import (
 // RequestReviewers creates a review request for the provided GitHub users for the specified pull request.
```
Comment (shurcooL) [inline github/pulls_reviewers.go]: > for the provided GitHub users

This was true before the change, but we'll need to update docs now to describe the new API.

File: github/pulls_reviewers.go [hunk 2]
```diff
@@ -13,18 +13,22 @@ import (
...
 //
 // GitHub API docs: https://developer.github.com/v3/pulls/review_requests/#create-a-review-request
-func (s *PullRequestsService) RequestReviewers(ctx context.Context, owner, repo string, number int, logins []string) (*PullRequest, *Response, error) {
+func (s *PullRequestsService) RequestReviewers(ctx context.Context, owner, repo string, number int, logins []string, teams []string) (*PullRequest, *Response, error) {
```
Comment (shurcooL) [inline github/pulls_reviewers.go]: @gmlewis, what do you think about this API decision?

An alternative to consider (that I can think of) is to create a `ReviewersRequest` struct which contains the two string slices, and have the user pass that in.

I'm pretty torn between the two, it seems they tradeoffs are pretty much equivalent and I'm not strongly in favor of either of them.

**Comment by shurcooL** [comment]
@RaviTezu Thanks for your work on this PR! It's looking very solid so far. I've only briefly looked at it. First, I want to think about the API and decide which direction we'll want to go. Afterwards, I can review the implementation details more closely.

**Comment by gmlewis** [comment]
LGTM
This seems like a good way to go (to me) for the API.
If @shurcooL comes up with a better alternative, I'm happy to discuss the tradeoffs.

**Comment by RaviTezu** [comment]
@shurcooL Any thoughts on how to move forward with this API?

**Comment by gmlewis** [comment]
@shurcooL's [suggestions](https://github.com/google/go-github/pull/682#pullrequestreview-54752451) SGTM.

**Comment by RaviTezu** [comment]
@shurcooL I agree with you.
It increases the readability and the recent commit has the suggested API changes. 
Take a look. @gmlewis

File: github/pulls_reviewers.go [hunk 3]
```diff
@@ -10,22 +10,31 @@ import (
...
+type Reviewers struct {
+	Users []*User `json:"users,omitempty"`
+	Teams []*Team `json:"teams,omitempty"`
+}
+
+// ReviewersRequest specifies users and teams for a pull request review request.
+type ReviewersRequest struct {
+	Reviewers     []string `json:"reviewers,omitempty"`
+	TeamReviewers []string `json:"team_reviewers,omitempty"`
+}
```
Comment (shurcooL) [inline github/pulls_reviewers.go]: Minor, swap the order of these two structs, so `ReviewersRequest`, the input, is on top, and `Reviewers`, the output, is on bottom.

Comment (RaviTezu) [inline github/pulls_reviewers.go]: @shurcooL Done.

**Pull request merged**